bitbots_blackboard/bitbots_blackboard/capsules/costmap_capsule.py

```diff
@@ -0,0 +1,390 @@
+             goal_value),
+            ((self.field_length, self.field_width / 2 + self.goal_width / 2 - goalpost_safety_distance),
+             goal_value),
+            ((self.field_length + self.map_margin,
+              self.field_width / 2 - self.goal_width / 2 - goalpost_safety_distance),
+             -0.2),
+            ((self.field_length + self.map_margin,
+              self.field_width / 2 + self.goal_width / 2 + goalpost_safety_distance),
+             -0.2),
+        ])
+
+        # Apply map margin to fixpoints
+        fix_points = [((p[0][0] + self.map_margin, p[0][1] + self.map_margin), p[1]) for p in fix_points]
+
+        # Interpolate the keypoints from above to form the costmap
+        interpolated = griddata([p[0] for p in fix_points], [p[1] for p in fix_points], (grid_x, grid_y),
+                                method='linear')
+
+        # Smooth the costmap to get more continus gradients
+        self.base_costmap = gaussian_filter(
+            interpolated,
+            self._blackboard.node.get_parameter(
+                "body.base_costmap_smoothing_sigma").value)
+        self.costmap = self.base_costmap.copy()
+
+    def get_gradient_at_field_position(self, x: float, y: float) -> Tuple[float, float]:
+        """
+        Gets the gradient tuple at a given field position
+        :param x: Field coordiante in the x direction
+        :param y: Field coordiante in the y direction
+        """
+        idx_x, idx_y = self.field_2_costmap_coord(x, y)
+        return -self.gradient_map[0][idx_x, idx_y], -self.gradient_map[1][idx_x, idx_y]
+
+    def get_cost_at_field_position(self, x: float, y: float) -> float:
+        """
+        Gets the costmap value at a given field position
+        :param x: Field coordinate in the x direction
+        :param y: Field coordinate in the y direction
+        """
+        idx_x, idx_y = self.field_2_costmap_coord(x, y)
+        return self.costmap[idx_x, idx_y]
+
+    def get_gradient_direction_at_field_position(self, x: float, y: float):
+        """
+        Returns the gradient direction at the given position
+        :param x: Field coordiante in the x direction
+        :param y: Field coordiante in the y direction
+        """
+        # for debugging only
+        #if False and self.costmap.sum() > 0:
+        #    # Create Grid
+        #    grid_x, grid_y = np.mgrid[0:self.field_length:self.field_length * 10j,
+        #                     0:self.field_width:self.field_width * 10j]
+        #    plt.imshow(self.costmap.T, origin='lower')
+        #    plt.show()
+        #    plt.quiver(grid_x, grid_y, -self.gradient_map[0], -self.gradient_map[1])
+        #    plt.show()
+
+        grad = self.get_gradient_at_field_position(x, y)
+        return math.atan2(grad[1], grad[0])
+
+    def get_cost_of_kick_relative(self, x: float, y: float, direction: float, kick_length: float, angular_range: float):
```

Comment:
Maybe add some interface docs to the last couple of functions.

Reply:
Can you check the interface docs? I added them

Reply:
Maybe clarify the angular range a bit. We use it to account for costs when we kick slightly left or right of our intended direction.

Reply:
Also maybe add to the x any description that they are relative to the robot.